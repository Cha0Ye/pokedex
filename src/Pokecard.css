.Pokecard{
    border: 2px solid black;
    background-color: lightsteelblue; 
    border-radius: 5px;
    width: 20vw;
    margin: 2.5vw;
    box-sizing: border-box;
}