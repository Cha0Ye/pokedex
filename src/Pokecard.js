import React, {Component} from 'react';

class Pokecard extends Component {
    
    render(){
        const { id, name, type, base_experience } = this.props;
        return (<div>
                    <h2> {name} </h2>
                    <img src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`} alt=''/>
                    <h3> {`Type: ${type}`}</h3>
                    <h3>EXP: <span>{base_experience}</span></h3>
               </div>);
    }

}

export default Pokecard;