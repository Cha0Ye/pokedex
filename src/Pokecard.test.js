import React from 'react';
import Pokecard from './Pokecard';
import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";

it("renders using shallow", function() {
    shallow(<Pokecard />);
  });

  it("renders using mount", function() {
    mount(<Pokecard />);
  });

  it("matches snapshot", function() {
    let wrapper = shallow(<Pokecard />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });