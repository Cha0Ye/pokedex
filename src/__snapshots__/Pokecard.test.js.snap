// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Pokecard"
>
  <h2>
     
     
  </h2>
  <img
    alt=""
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
  />
  <h3>
     
    Type: undefined
  </h3>
  <h3>
    EXP: 
    <span />
  </h3>
</div>
`;
